src/test/java/com/epam/aidial/core/ProxyTest.java

```diff
@@ -184,6 +191,37 @@ public void testHandle_UnknownApiKey() {
         verify(response).setStatusCode(UNAUTHORIZED.getCode());
     }
 
+    @Test
+    public void testHandle_SuccessApiKey() {
+        when(request.version()).thenReturn(HttpVersion.HTTP_1_1);
+        when(request.method()).thenReturn(HttpMethod.GET);
+        MultiMap headers = mock(MultiMap.class);
+        when(request.headers()).thenReturn(headers);
+        when(request.getHeader(eq(HttpHeaders.CONTENT_TYPE))).thenReturn(null);
+        when(headers.get(eq(HEADER_API_KEY))).thenReturn("key1");
+        when(headers.get(eq(HttpHeaders.CONTENT_LENGTH))).thenReturn(Integer.toString(512));
+        when(request.path()).thenReturn("/foo");
+        when(request.uri()).thenReturn("/foo");
+
+        Config config = new Config();
+        config.setKeys(Map.of("key1", new Key()));
+        Route route = new Route();
+        route.setMethods(Set.of(HttpMethod.GET));
+        route.setName("route");
+        route.setPaths(List.of(Pattern.compile("/foo")));
+        route.setResponse(new Route.Response());
+        LinkedHashMap<String, Route> routes = new LinkedHashMap<>();
+        routes.put("route", route);
+        config.setRoutes(routes);
+        when(configStore.load()).thenReturn(config);
+
+        when(accessTokenValidator.extractClaims(any())).thenReturn(Future.succeededFuture(IdentityProvider.CLAIMS_WITH_EMPTY_ROLES));
+
+        proxy.handle(request);
+
+        verify(response).setStatusCode(OK.getCode());
```

Comment:
So the only things we are testing is that we are getting OK? 

Is there any testing for the main functionality of this PR? That app/adapter actually get generated keys which are different to the key request was originally authorized with? And that they actually can use these generated keys?

Reply:
there is already a such test com.epam.aidial.core.ProxyTest#testHandle_UnknownApiKey

Reply:
I just add a new test with the success path.

Reply:
fixed